### DISCUSSION TITLE
Can I add new shortcut keys?

### DISCUSSION THREAD

**Discussion by gorimatyan** [facebook/lexical#3750] [Q&A] [answered]
I want to know how to add shortcut keys such as  **[ ctr + [any key] ]**.
In other discussion, I understood adding [ctr + Enter] shortcut keys.
https://github.com/facebook/lexical/discussions/1805

Though I looked up lexical/src/LexicalCommands.ts, this file didn't define commands I want.
So I want to create new commands such as [ctr+s (overwrite saving file behavior and add new behavior)], [ctr + t] and so on.
The behaviors of these shortcut keys ,I expect, is adding underline to selected text or changing color of text when I press [ctr + [any key]].

**Comment by moy2010** [comment, marked as answer]
You can do it like this:

```
const [editor] = useLexicalComposerContext();

useEffect(() => {
  return mergeRegister(
    editor.registerRootListener((rootElement) => {
      if (rootElement) {
        rootElement.addEventListener('keydown', yourKeyDownHandler);
      }
    })
  )
}, [editor, editorReadCallback]);
```

**Reply by trueadm** [reply]
That will register an event listener on each update. You should probably be using `registerRootListener` and in there you can register your event listener.

**Reply by moy2010** [reply]
Ah, thx. @trueadm ! Will update the example above.

**Reply by gorimatyan** [reply]
Thank you for answering. I'll do that.

**Reply by gorimatyan** [reply]
@moy2010 @trueadm 
Excuse me.
I don't know your using mergeRegister(). What does it is used for ??
According to the doc, mergeRegister() defined in lexical-utils/src/index.ts:232.

```
export function mergeRegister(...func: Array<Func>): () => void {
  return () => {
    func.forEach((f) => f());
  };
}
```

**Reply by moy2010** [reply]
Well, yeah. If you are only registering one listener, then there's no need for the `mergeRegister` call :smile: .

**Reply by gorimatyan** [reply]
I got it!! Thank you!!
